packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -594,26 +689,78 @@ class _CupertinoNavigationBarState extends State<CupertinoNavigationBar> {
       userMiddle: widget.middle,
       userTrailing: widget.trailing,
       padding: widget.padding,
-      userLargeTitle: null,
-      large: false,
+      userLargeTitle: widget.largeTitle,
+      large: widget.largeTitle != null,
+      staticBar: true, // This one does not scroll
     );
 
-    final Widget navBar = _wrapWithBackground(
+    Widget navBar = Column(
+      children: [
+        _PersistentNavigationBar(
+          components: components,
+          padding: widget.padding,
+          middleVisible: widget.middle != null,
+        ),
+        if (widget.bottom != null) widget.bottom!,
+      ],
+    );
+
+    if (widget.largeTitle != null) {
+      navBar = ConstrainedBox(
+        constraints: BoxConstraints(maxHeight: maxHeight),
+        child: Stack(
```

Comment:
I think the `Stack` was there for the expanding/collapsing behavior? If its just static then only the `DefaultTextStyle ` subtree should remain?

Reply:
Ah I see the static navbar is in the `Stack`. But wouldn't that cause problems if the `bottom` is provided? If the large title is entirely static, shouldn't it just sit in the static nav bar column above the `bottom`?

Reply:
Example of what it looks like if a `bottom` is provided:

<details>
<summary>Image</summary>

![Simulator Screenshot - iPhone 15 Pro - 2024-10-18 at 11 31 51](https://github.com/user-attachments/assets/f246274c-0486-47f5-880e-e8f23e9aafea)

</details>

<details>
<summary>Sample Code</summary>

```dart
// Copyright 2014 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import 'package:flutter/cupertino.dart';

/// Flutter code sample for [CupertinoNavigationBar.large].
void main() {
  runApp(const NavBarApp());
}


class NavBarApp extends StatelessWidget {
  const NavBarApp({super.key});

  @override
  Widget build(BuildContext context) {
    return const CupertinoApp(
      theme: CupertinoThemeData(brightness: Brightness.light),
      home: NavBarExample(),
    );
  }
}

class NavBarExample extends StatefulWidget {
  const NavBarExample({super.key});

  @override
  State<NavBarExample> createState() => _NavBarExampleState();
}

class _NavBarExampleState extends State<NavBarExample> {
  int _count = 0;

  @override
  Widget build(BuildContext context) {
    return CupertinoPageScaffold(
      navigationBar: const CupertinoNavigationBar.large(
        largeTitle: Text('Large Sample'),
        bottom: PreferredSize(
            preferredSize: Size.fromHeight(50), child: Placeholder()),
      ),
      child: SafeArea(
        child: Center(
          child: Column(
            children: <Widget>[
              const Spacer(),
              const Text('You have pushed the button this many times:'),
              Text(
                '$_count',
                style:
                    CupertinoTheme.of(context).textTheme.navLargeTitleTextStyle,
              ),
              const Spacer(),
              Padding(
                padding: const EdgeInsets.all(15.0),
                child: CupertinoButton.filled(
                  onPressed: () => setState(() => _count++),
                  child: const Text('Increment'),
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }
}
```

</details>

Reply:
Ah thank you! Bottom was added while I was working on this, will update!

Reply:
Since it's static, we would expect to have a column of persistent nav bar -> Large title (if provided) -> bottom (if provided). In that case, what's the purpose of the `Stack` vs. just throwing the large title into the column if the `.large` constructor is used?

Reply:
Ah ok, I misunderstood then, I thought the persistent components had no background and were placed on top of the large header widget. I will adjust that.
I noticed I introduced a regression in one of the CupertinoSliverNavigationBar demo apps in the golden file test. Will clean both of these up now, thanks @victorsanni!